apps/bplan/serializers.py

```diff
@@ -104,12 +102,35 @@ def _get_absolute_url(self, bplan):
         return url
 
     def _download_image_from_url(self, url):
-        parsed_url = urlparse(url)
-        file_path = os.path.join(PROJECT_IMAGE_DIR,
-                                 os.path.basename(parsed_url.path))
-        file_name = os.path.join(settings.MEDIA_ROOT, file_path)
-        file_dir = os.path.dirname(file_name)
-        os.makedirs(file_dir, exist_ok=True)
-
-        request.urlretrieve(url, file_name)
+        try:
+            parsed_url = urlparse(url)
+            file_path = os.path.join(PROJECT_IMAGE_DIR,
+                                     os.path.basename(parsed_url.path))
+            file_name = os.path.join(settings.MEDIA_ROOT, file_path)
+            file_dir = os.path.dirname(file_name)
+            os.makedirs(file_dir, exist_ok=True)
+
+            request.urlretrieve(url, file_name)
+        except:
+            self._remove_image_if_exists(file_name)
+            raise serializers.ValidationError(
+                'Failed to download image {}'.format(url))
+        self._validate_image(file_name)
         return file_path
+
+    def _validate_image(self, file_name):
+        image_file = open(file_name, "rb")
+        image = ImageFile(image_file, file_name)
+        config = settings.IMAGE_ALIASES.get('*', {})
+        config.update(settings.IMAGE_ALIASES['heroimage'])
+        try:
+            validate_image(image, **config)
+        except ValidationError as e:
+            self._remove_image_if_exists(file_name)
```

Comment:
i don't like the idea, that the validator tries to delete the validated image. see above